google/cloud/aiplatform/lro.py

```diff
@@ -15,6 +15,46 @@
 # limitations under the License.
 #
 
+from typing import Optional
+
+from google.api_core import operation as ga_operation
+from google.cloud.aiplatform import base
+
 
 class LRO:
-    pass
+    """A handler for operation futures."""
+
+    def __init__(
+        self,
+        operation: ga_operation.Operation,
+        resource_noun_obj: Optional[base.AiPlatformResourceNoun] = None,
+    ):
+        """Initialises class with operation and optional object to update.
+
+        Args:
+            operation (ga_operation.Operation): operation to handle
+        """
+        self._operation_future = operation
+        if resource_noun_obj:
+            self.add_update_resource_callback(resource_noun_obj)
+
+    def add_update_resource_callback(
+        self, resource_noun_obj: base.AiPlatformResourceNoun
+    ):
+        """Updates resource with result of operation.
+
+        Args:
+            resource_noun_obj (base.AiPlatformResourceNoun):
+                resource to be updated upon operation completion
+        """
+
+        def callback(operation_future):
+            result_obj = operation_future.result()
+            resource_noun_obj._gca_resource = result_obj
```

Comment:
The result of the operation_future may not be the one we want to assign to _gca_resource. For example operation_future.result for upload_model returns a UploadModelResponse: https://github.com/googleapis/python-aiplatform/blob/master/google/cloud/aiplatform_v1beta1/services/model_service/client.py#L192

That response has the resource name for the model but it's not the Model resource proto we need to assign to _gca_resource which is here: https://github.com/googleapis/python-aiplatform/blob/master/google/cloud/aiplatform_v1beta1/types/model.py#L34

We have to make an additional API call to get that proto. That's why this method should look something like this so we can pass in the get resource proto API call to be executed when the operation is complete:

```python
def add_update_resource_callback(self,
   resource_noun_obj: base.AiPlatformResourceNoun,
   result_key: str, api_get: Callable):
   def callback(result):
     result_value = getattr(operation_future.result(), result_key)
     resource = api_get(name=result_value)
     resource_noun_obj._gca_resource = resource
  self._operation_future.add_done_callback(callback)
```

Usage from Model would look like this:

```python
lro.add_update_resource_callback(self, 'model', api_client.get_model)
```

Reply:
ah i didn't account for this scenario. this makes sense.